freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UmsUsersStateProvider.java

```diff
@@ -43,6 +43,8 @@
 
     private static final String ADMIN_FREEIPA_GROUP = "admins";
 
+    private static final boolean INCLUDE_INTERNAL_MACHINE_USERS = true;
```

Comment:
Don't we want to make this configurable?

Reply:
We do not want to make this configurable. We always want to include internal machine users. Internal machine users may be identities within the environment. Right now, single user syncs that provide a machineUser crn can sync the user to the environment but then the users will be deleted by a full sync. This is incorrect behavior that we need to fix.

I don't think there are any internal machineUsers currently in use, but the DWX team is working on creating an internal machine user for their service and potentially the DEX team.

I'll update the commit message to include more details about the behavior we are fixing